diff --git a/CMakeLists.txt b/CMakeLists.txt
index 808330e..8adb110 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,11 @@ if (POLICY CMP0063)
   cmake_policy (SET CMP0063 NEW)
 endif (POLICY CMP0063)
 
+# option() honors normal variables.
+if (POLICY CMP0077)
+  cmake_policy (SET CMP0077 NEW)
+endif (POLICY CMP0077)
+
 project(glog VERSION 0.4.0 LANGUAGES C CXX)
 
 set (CPACK_PACKAGE_NAME glog)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 808330e..49ed082 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -394,6 +394,10 @@ elseif (UNIX OR (APPLE AND HAVE_DLADDR))
   set (HAVE_SYMBOLIZE 1)
 endif (WIN32 OR CYGWIN)
 
+if (EMSCRIPTEN)
+  set (HAVE_SYMBOLIZE 0)
+endif (EMSCRIPTEN)
+
 check_cxx_source_compiles ("
 #include <cstdlib>
 #include <time.h>
